pkgs/by-name/ve/vep/package.nix

```
@@ -0,0 +1,122 @@
+# Instructions:http://www.ensembl.org/info/docs/api/api_installation.html
+# Do not use https://github.com/Ensembl/ensembl-vep/archive/release/${version}.zip
+# We cannot use INSTALL.pl but it’s not that bad:install the dependencies and copies the .pm files should be ok
+{
+  lib,
+  htslib,
+  perlPackages,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  perl,
+  makeWrapper,
+}:
+
+let
+  version = "110";
+  customInstallPhase = ''
+    mkdir -p $out/${perl.libPrefix}/${perl.version}/
+    tests=$(find modules/ -mindepth 1 -maxdepth 1 -type d | grep -v t)
+    cp -r $tests $out/${perl.libPrefix}/${perl.version}/
```

Custom phases need to run hooks:

```suggestion
    runHook preInstall
    
    mkdir -p $out/${perl.libPrefix}/${perl.version}/
    tests=$(find modules/ -mindepth 1 -maxdepth 1 -type d | grep -v t)
    cp -r $tests $out/${perl.libPrefix}/${perl.version}/
    
    runHook postInstall
```

---

:point_up:

---

I still don't see hooks here. You've added them to a copy of this script, but kept this script hookless? What's going on?

---

Thanks for catching that up. I've added the hooks.